### ISSUE TITLE
Strange error

### ISSUE THREAD

**Issue by tgiachi** [JasperFx/marten#938]
Hi, I've strange (random timing)

Psql server is up and healty!

Thanks!
Tommy
`Marten.MartenCommandException: Marten Command Failure:
      

      Exception while writing to stream ---> Npgsql.NpgsqlException: Exception while writing to stream ---> System.ObjectDisposedException: Cannot access a disposed object.
      Object name: 'System.Net.Sockets.NetworkStream'.
         at System.Net.Sockets.NetworkStream.Write(Byte[] buffer, Int32 offset, Int32 size)
         at Npgsql.WriteBuffer.<Flush>d__27.MoveNext()
         --- End of inner exception stack trace ---
         at Npgsql.WriteBuffer.<Flush>d__27.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Npgsql.NpgsqlCommand.<SendExecute>d__78.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Npgsql.NpgsqlCommand.<Execute>d__71.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
         at System.Runtime.CompilerServices.ValueTaskAwaiter`1.GetResult()
         at Npgsql.NpgsqlCommand.<ExecuteDbDataReader>d__92.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
         at System.Runtime.CompilerServices.ValueTaskAwaiter`1.GetResult()
         at Npgsql.NpgsqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
         at Marten.Schema.SchemaPatch.Apply(NpgsqlConnection conn, AutoCreate autoCreate, ISchemaObject[] schemaObjects)`

**Comment by jeremydmiller** [comment]
I don't know that that's anything Marten specific. If you're getting this from schema patching, you can try using AutoCreate.None to stop that.

I'm closing just out of no recent activity.

**Comment by malscent** [comment]
I had the exact same issue when executing multiple tests in xUnit testing some services that utilized a UnitOfWork on top of the DocumentStore's Sessions.

My culprit was that I was trying to keep the lifecycle of the IDocumentSession too long, and it was being Collected causing the PG Connection to drop.  I needed to create a Unit of work within a using, perform actions and then dispose of it, instead of trying to maintain the same unit of work across the lifecycle of a service.

I created a repo to demonstrate the issue here:  

https://github.com/malscent/MartenHelp

Hope that helps anyone who finds this issue like I did.
